go/runtime/registry/registry.go

```diff
@@ -124,6 +133,10 @@ func (r *runtime) WatchRegistryDescriptor() (<-chan *registry.Runtime, pubsub.Cl
 	return ch, sub, nil
 }
 
+func (r *runtime) RegisterStorage(storage storageAPI.Backend) {
+	r.storage = storage
```

Comment:
Should this now be taking `r.Lock()`?

Reply:
Also should this panic if storage was already registered? To prevent accidental overwrites.